addons/web/static/tests/views/calendar_tests.js

```diff
@@ -844,6 +847,46 @@ QUnit.module('Views', {
         calendar.destroy();
     });
 
+    QUnit.test('render popover with widget which has specialData attribute', async function (assert) {
+        assert.expect(2);
+
+        await testUtils.mock.patch(BasicModel, {
+            _fetchSpecialData() {
```

Comment:
should be on specialWidget --> no need to patch BasicModel

Reply:
@Polymorphe57 I don't understand if BasicModel is not patched then what should be patched?

Reply:
the function should be defined in MyWidget

Reply:
@Polymorphe57 function defined as ```specialData``` attribute is written in Model, how could I define it in widget?

Reply:
look for
`            if (Widget && Widget.prototype.specialData) {
                return self[Widget.prototype.specialData](record, name, fieldInfo).then(function (data) {`
in basic_model.js

Reply:
@Polymorphe57 Please check here we are calling method of self, i.e. method of BasicModel

Reply:
Ok. So you should override _fetchSpecialAttendeeStatus and define it as specialData value in your widget. That  would be better. We should not override _fetchSpecialData.

Reply:
@Polymorphe57 I would add ```_fetchSpecialAttendeeStatus``` as a ```specialData``` attribute but ```_fetchSpecialAttendeeStatus``` is deined in calendar module and test is defined in web module, that's why I added custom ```specialData``` here, the goal of the test here is to test whether ```specialData``` method is called when such widget is available in calendar popover, what do you say?

Reply:
Ok, I agree with the principle but here _fetchSpecialData is called only once (due to the way you override the function) and you don't test what you want. So you should call super here. In that case, we'll be fine.

Reply:
@Polymorphe57 Let me clear this ```specialData``` attribute, if I am introducing any ```specialData``` attribute in any widget then that method will be introduced in BasicModel, so here in test I have introduced ```specialData: "_fetchSpecialData",``` so it means I will have to define ```_fetchSpecialData``` in BasicModel, so here in this case I define ```_fetchSpecialData``` method in BasicModel by patching it, We don't need to call super as super class will not have ```_fetchSpecialData``` as it is intorduced by me in BasicModel by patching it, let me know if it is still not clear.

Reply:
Hello Mohammed. If I am not wrong _fetchSpecialData is already defined in basic_model.js. This is where the method  that you refer to with the key/value pair specialData: _fetchSpecialData will be called. So super already exists and is called in the code of basic_model.js (see https://github.com/odoo/odoo/blob/master/addons/web/static/src/js/views/basic/basic_model.js#L2712)

Reply:
and https://github.com/odoo/odoo/blob/master/addons/web/static/src/js/views/basic/basic_model.js#L4291 for example

Reply:
Am I really missing something?

Reply:
I mean, _fetchSpecialData will be called anyway by the basic model, then it should be called once again because you refer to it in the definition of your widget. To make things clearer, we could maybe define another method (that does not exists anywhere) in BasicModel and define it as the value of specialData?

Reply:
@Polymorphe57 Finally I agree :) the name creates confusion, actually Ishould give different name to ```specialData``` value as ```_fetchSpecialData``` already there in basic_model.js, Thanks